SubLinkCommon/SubLinkService.cs

```diff
@@ -18,28 +20,53 @@ public class SubLinkService<TGlobals, TCompilerService, TService> : BackgroundSe
     private readonly ILogger _logger;
     private readonly IServiceScopeFactory _serviceScopeFactory;
 
-    public SubLinkService(ILogger logger, IServiceScopeFactory serviceScopeFactory) {
+    // ReSharper disable once PrivateFieldCanBeConvertedToLocalVariable
+    private readonly IOptionsMonitor<SubLinkSettings> _settingsMonitor;
+    private SubLinkSettings _settings;
+
+    public SubLinkService(
+        ILogger logger,
+        IServiceScopeFactory serviceScopeFactory,
+        IOptionsMonitor<SubLinkSettings> settingsMonitor
+    ) {
         _logger = logger;
         _serviceScopeFactory = serviceScopeFactory;
+
+        _settingsMonitor = settingsMonitor;
+        _settingsMonitor.OnChange(UpdateSubLinkSettings);
+        _settings = _settingsMonitor.CurrentValue;
```

Comment:
If introducing mutability is required then it should ideally be constrained as much as possible, or explained clearly what the expected scope of the mutability should be and what behavior is desired/what is not desired.

If you feel strongly about this then please add a comment in the code justifying it.